src/services/invokeService.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseService } from "./baseService"
+import Serverless from "serverless";
+import config from "../config";
+import axios from "axios";
+import { FunctionAppService } from "./functionAppService";
+import { FunctionAppResource } from "../armTemplates/resources/functionApp";
+import { ServerlessAzureConfig } from "../models/serverless";
+
+export class InvokeService extends BaseService {
+  public serverless: Serverless;
+  public options: Serverless.Options;
+  public constructor(serverless: Serverless, options: Serverless.Options) {
+    super(serverless, options);
+    this.serverless = serverless;
+    this.options = options;
+  }
+
+  /**
+   * Invoke an Azure Function
+   * @param functionName Name of function to invoke
+   * @param data Data to use as body or query params
+   * @param method GET or POST
+   */
+  public async invoke(functionName: string, data: any, method: string){
+    /* accesses the admin key */
+    if (!(functionName in this.slsFunctions())) {
+      this.serverless.cli.log(`Function ${functionName} does not exist`);
+      return;
+    }
+    const functionObject = this.slsFunctions()[functionName];
+    const eventType = Object.keys(functionObject["events"][0])[0];
+    if (eventType !== "http") {
+      this.log("Needs to be an http function");
+      return;
+    }
+
+    let url = `http://${FunctionAppResource.getResourceName(this.serverless.service as any)}${config.functionAppDomain}${config.functionAppApiPath + functionName}`;
```

Comment:
`config.functionAppDomain` from the constants is not correct in all cases.  I would use the function app service to get to domain from the actual function app configuration.  Also - I would remove dependency on the `config` constants all together since in this updated model they are not really used any more.  Take a look @ `FunctionAppService.getFunctionHttpTriggerConfig` for more info.

The `functionName` may not always be used as the api endpoint.  This is configurable as part of the binding.